csv_parser/test_case_generator.py

```diff
@@ -95,8 +161,49 @@ def run(perimeters):
     return test_cases
 
 
+def generate_step_json(perimeter, test_case, row, receive_json):
+    # Create the JSON file for the step
+    if not os.path.exists(f'./out/test-cases/{perimeter["name"]}/{test_case["label"]}'):
+        os.makedirs(f'./out/test-cases/{perimeter["name"]}/{test_case["label"]}')
+    with open(
+            f'./out/test-cases/{perimeter["name"]}/{test_case["label"]}/{len(test_case["steps"])}-{row["Pas de test"]} {row["Modèle"]}.json',
+            'w', encoding='utf-8') as file:
+        file.write(dumps(receive_json, indent=4, ensure_ascii=False))
+
+
 def get_type(type_in_french):
     if type_in_french == "Envoi":
         return "send"
     elif type_in_french == "Réception":
         return "receive"
+
+
+def generate_type_override_map(row):
+    # We load the relevant schema from ../src/main/resources/json-schema/row["Pas de test"].schema.json into the
+    # schema object
+    schema = json.load(open(f'../src/main/resources/json-schema/{row["Modèle"]}.schema.json'))
+    # We iterate over all properties recursively in the schem and add those that have simple types different from
+    # string (i.e. integer, number, boolean) to the type_override_map object
+    type_override_map = {}
+    for key, value in nested_dict_iter(schema["definitions"]):
+        if "type" in value and not isinstance(value["type"], collections.Mapping) and value["type"] not in {"string",
+                                                                                                            "object", "array"}:
+            type_override_map[key] = value["type"]
+    return type_override_map
+
+
+def nested_dict_iter(nested):
+    for key, value in nested.items():
+        if isinstance(value, collections.Mapping):
+            yield key, value
+            for inner_key, inner_value in nested_dict_iter(value):
+                yield inner_key, inner_value
```

Comment:
à quoi sert cette fonction ?

Reply:
Ca permet d'iterer dans les definitions/proprietes imbriquées et recuperer en meme temps chaque de ces definitons/propriete.

Reply:
collections.Mapping devient déprécié a priori : https://github.com/davidhalter/parso/issues/76
Je pense qu'il faudrait fixer la version de python dans la github action aussi d'ailleurs pour éviter les surprises quand elle change dans ubuntu-latest

Reply:
Mais très élégant, je ne connaissais pas yield en fait, assez puissant pour créer un generateur pour itérer comme on souhaite !
Et pourquoi c'est pas plutôt ?
```python
def nested_dict_iter(nested):
    for key, value in nested.items():
        yield key, value
        if isinstance(value, collections.Mapping):
            for inner_key, inner_value in nested_dict_iter(value):
                yield inner_key, inner_value
`
``
Je comprends pas comment ton code arrive à retourner les propriétés pas objets à la racine {'prop': 'elle', ...} par exemple